block/s3/inventory.go

```diff
@@ -77,6 +81,36 @@ func (inv *Inventory) InventoryURL() string {
 	return inv.Manifest.URL
 }
 
+func readFileMetadata(ctx context.Context, logger logging.Logger, s3 s3iface.S3API, getParquetReader parquetReaderGetter, m *manifest) error {
+	m.numRowsByFilename = make(map[string]int, len(m.Files))
+	m.firstKeyByFilename = make(map[string]string, len(m.Files))
+	for i := range m.Files {
+		filename := m.Files[i].Key
+		pr, closeReader, err := getParquetReader(ctx, s3, m.invBucket, filename)
+		if err != nil {
+			return err
+		}
+		m.numRowsByFilename[filename] = int(pr.GetNumRows())
+
+		// read first row from file to store the first key:
+		rows := make([]ParquetInventoryObject, 1)
+		err = pr.Read(&rows)
+		if err != nil {
+			return err
+		}
+		m.firstKeyByFilename[filename] = ""
+		if len(rows) != 0 {
+			m.firstKeyByFilename[filename] = rows[0].Key
+		}
+		err = closeReader()
+		if err != nil {
+			logger.WithFields(logging.Fields{"bucket": m.invBucket, "key": filename}).
+				Error("failed to close parquet reader in readFileMetadata")
```

Comment:
usually I do no log the function name as part of the method - can have it as part of the key fields. but in this case I think our logger already capture the error location.
maybe "failed to close parquet while generate inventory"